Frontend/nutri-pal/src/components/mealRegister/MealRegistrationForm.tsx

```diff
@@ -0,0 +1,94 @@
+'use client'
+
+import '@/styles/MealRegistrationForm.css'
+import { BASE_URL_REST_API } from '@/app/consts'
+import axios from 'axios'
+import { auth } from '@/app/firebase-config'
+import { onAuthStateChanged } from 'firebase/auth'
+import { useState } from 'react'
+
+interface Values {
+    name?: string,
+    protein?: number,
+    carbs?: number,
+    fats?: number,
+    kcal?: number,
+}
+
+export const MealRegistrationForm = ({ name, protein, carbs, fats, kcal }: Values) => {
+    const [user, setUser] = useState({})
+
+    // Gets the information of the user that is signed in
+    onAuthStateChanged(auth, (user) => {
+        if (user) {
+            setUser(user);
+        }
+    })
+
+    // When the form is submitted then do a POST request to the REST API of Meals
+    const sendMeal = (e: any) => {
+        e.preventDefault();
+
+        const data = new FormData(e.currentTarget);
+        const values = Object.fromEntries(data.entries());
+
+        axios.post(BASE_URL_REST_API + 'meals', {
+            name: values.name,
+            category: values.category,
+            description: values.description,
+            proteins: parseInt(values.protein),
+            carbs: parseInt(values.carbs),
+            fats: parseInt(values.fats),
+            calories: parseInt(values.calories),
+            userId: user.uid
+        })
+    }
+
+    return (
```

Comment:
this form is ginormous we can break this up into smaller components similar to the suggestion above

```
{nameInput}
{descriptionInput}
{categoryDropdown}
{nutritionInputs}
```

or

```
{nameInput}
{descriptionInput}
{categoryDropdown}
<div className='meal-register-form-grams-input-container'>
     {proteinInput}
     {fatInput}
     ...
</div>
{submitButton}
```